cmd/register/register_util.go

```diff
@@ -261,29 +283,35 @@ func DownloadFileFromHTTP(ctx context.Context, ref storage.DataReference) (io.Re
 }
 
 /*
-Get file list from the args list.
-If the archive flag is on then download the archives to temp directory and extract it.
+Get serialize output file list from the args list.
+If the archive flag is on then download the archives to temp directory and extract it. In case of fast register it will also return the compressed source code
 The o/p of this function would be sorted list of the file locations.
 */
-func getSortedFileList(ctx context.Context, args []string) ([]string, string, error) {
+func getSerializeOutputFiles(ctx context.Context, args []string) ([]string, string, error) {
 	if !rconfig.DefaultFilesConfig.Archive {
 		/*
 		 * Sorting is required for non-archived case since its possible for the user to pass in a list of unordered
 		 * serialized protobuf files , but flyte expects them to be registered in topologically sorted order that it had
 		 * generated otherwise the registration can fail if the dependent files are not registered earlier.
 		 */
+		for _, v := range args {
+			if (len(rconfig.DefaultFilesConfig.AdditionalDistributionDir) == 0 || len(rconfig.DefaultFilesConfig.DestinationDir) == 0) && strings.HasSuffix(v, sourceCodeExtension) {
+				return args, "", fmt.Errorf("you are trying to register fast serialize workflow. Please pass additional flags like --additionalDistributionDir and --destinationDir")
+			}
+		}
```

Comment:
This logic need not be here and can be part of separate areFastRegisterFlagsPassed() and can simply be done once we identify if the archive contains the fast serialized sources